tests/integration/utils.py

```diff
@@ -242,62 +268,88 @@ def assert_content_manifest_schema(response_data):
+    :param files_to_ignore: The list of files which can be ignored in the assertion
     """
-    if expected_file_urls is None:
-        expected_file_urls = {}
-    assert os.path.exists(source_path) and os.path.isdir(
-        source_path
-    ), f"Invalid source_path: {source_path}"
-    files = []
-    # Go through all files in source_code_path and it's subdirectories
-    for root, _, source_files in os.walk(source_path):
-        for file_name in source_files:
-            # Get path to file in the project
-            absolute_file_path = os.path.join(root, file_name)
-            relative_file_path = os.path.relpath(absolute_file_path, start=source_path)
-            # Assert that content of source file is equal to expected
-            with open(absolute_file_path, "rb") as f:
-                if check_content:
-                    # Download expected file
-                    file_url = expected_file_urls[relative_file_path]
-                    expected_file = requests.get(file_url).content
-                    assert f.read() == expected_file
-                else:
-                    assert os.path.isfile(
-                        absolute_file_path
-                    ), f"File {absolute_file_path} does not exist"
-            files.append(relative_file_path)
-    # Assert that there are no missing or extra files
-    assert set(files) == set(list(expected_file_urls))
+    for dir_to_check, archive_url in expected_files.items():
+        # A directory path to check
+        test_path = os.path.join(source_path, dir_to_check)
+        # If there is no link to expected archive, the directory should be empty
+        if not archive_url:
+            assert (
+                len(os.listdir(test_path)) == 0
+            ), f"Directory: {test_path} not found or not empty as expected."
+        else:
+            # A directory path with extracted deps
+            deps_data_path = os.path.join(tmpdir, "test_source")
+            # An archive path with expected data
+            expected_archive = os.path.join(tmpdir, "archive.tar.gz")
+            # A directory path with extracted expected data
+            expected_data_path = os.path.join(tmpdir, "expected_data")
+            # Dependencies instead of source code are saved as archives.
+            # Therefore we should extract it firstly
+            if dir_to_check.startswith("deps"):
+                archive_path = test_path
+                extract_archive(archive_path, deps_data_path)
+                package_root_dir = os.path.join(deps_data_path, os.listdir(deps_data_path)[0])
+            else:
+                package_root_dir = test_path
+            download_archive(archive_url, expected_archive)
+            extract_archive(expected_archive, expected_data_path)
+            # Root directory for expected data of package or dependency
+            expected_package_root_dir = os.path.join(
+                expected_data_path, os.listdir(expected_data_path)[0]
+            )
```

Comment:
Oh, this might be the reason why it works, the problem is you're likely checking the same extracted archive in each iteration (but `os.listdir` does not specify order, so who knows).